common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1872,25 +1870,72 @@ func (d *cassandraPersistence) ResetWorkflowExecution(request *p.InternalResetWo
 		startVersion = insertReplicationState.StartVersion
 		lastWriteVersion = insertReplicationState.LastWriteVersion
 	}
-	batch.Query(templateUpdateCurrentWorkflowExecutionQuery,
-		insertExecutionInfo.RunID,
-		insertExecutionInfo.RunID,
-		insertExecutionInfo.CreateRequestID,
-		insertExecutionInfo.State,
-		insertExecutionInfo.CloseStatus,
-		startVersion,
-		lastWriteVersion,
-		lastWriteVersion,
-		insertExecutionInfo.State,
-		d.shardID,
-		rowTypeExecution,
-		insertExecutionInfo.DomainID,
-		insertExecutionInfo.WorkflowID,
-		permanentRunID,
-		defaultVisibilityTimestamp,
-		rowTypeExecutionTaskID,
-		request.PrevRunID,
-	)
+
+	if currReplicationState == nil {
+		batch.Query(templateUpdateCurrentWorkflowExecutionQuery,
+			insertExecutionInfo.RunID,
+			insertExecutionInfo.RunID,
+			insertExecutionInfo.CreateRequestID,
+			insertExecutionInfo.State,
+			insertExecutionInfo.CloseStatus,
+			startVersion,
+			lastWriteVersion,
+			lastWriteVersion,
+			insertExecutionInfo.State,
+			d.shardID,
+			rowTypeExecution,
+			insertExecutionInfo.DomainID,
+			insertExecutionInfo.WorkflowID,
+			permanentRunID,
+			defaultVisibilityTimestamp,
+			rowTypeExecutionTaskID,
+			currExecutionInfo.RunID,
+		)
+	} else {
+		// NOTE: workflow_state is either 1 or 2, there is no 0(created)
+		prevRunState := 2
```

Comment:
i think this value should be passed in, btw, plz use WorkflowStateRunning or WorkflowStateCompleted

Reply:
OK.